pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java

```diff
@@ -1149,16 +1165,17 @@ public int downloadFile(URI uri, File dest, AuthProvider authProvider)
    * Download a file.
    *
    * @param uri URI
+   * @param socketTimeoutMs socketTimeoutMs
    * @param dest File destination
    * @param authProvider auth provider
    * @param httpHeaders http headers
    * @return Response status code
    * @throws IOException
    * @throws HttpErrorStatusException
    */
-  public int downloadFile(URI uri, File dest, AuthProvider authProvider, List<Header> httpHeaders)
+  public int downloadFile(URI uri, int socketTimeoutMs, File dest, AuthProvider authProvider, List<Header> httpHeaders)
```

Comment:
Let's keep this signature as is, and add a new one which takes `socketTimeoutMs`